packages/core/server/admin-api/services/query-layer-decorator.ts

```diff
@@ -250,6 +143,69 @@ const decorator = (service: IDecoratedService) => ({
     // Eventually delete the entity.
     return service.delete.call(this, uid, entityId);
   },
+
+  async clone(
+    uid: Common.UID.ContentType, cloneId: number, params?: IDecoratedServiceOptions<{ url_alias: number }>
+  ) {
+    const hasWT = isContentTypeEnabled(uid);
+    if (!hasWT) {
+      return service.clone.call(this, uid, cloneId, params);
+    }
+
+    // Manually fetch the entity to clone
+    const entityToClone = await service.findOne.call(this, uid, cloneId, { populate: ["url_alias"] });
+
+    if (!entityToClone) {
+      throw new Error('Entity to clone not found');
+    }
+
+    // Remove the id and url_alias from the entity to clone
+    const { id, url_alias, ...clonedData } = entityToClone;
+
+    // Create the cloned entity
+    const clonedEntity = await service.create.call(this, uid, { data: clonedData });
+
+    if (!clonedEntity) {
+      throw new Error('Cloning failed, cloned entity is null or undefined');
+    }
+
+    // Handle URL alias for the cloned entity
+    if (url_alias) {
+      const newUrlAlias = await getPluginService('urlAliasService').create({
+        url_path: `${url_alias.url_path}-clone`,
```

Comment:
Couple remarks here.

1. I think we shouldn't append `-clone` to the new url_path.
2. You're now using the `url_alias.url_path` which is the same path as the source entity. Though in theory a user could change the values before performing the clone.

So I think we should regenerate the url path here, based on the `clonedEntity`.